controllers/alerts/secret.go

```diff
@@ -0,0 +1,97 @@
+package alerts
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kubevirt/hyperconverged-cluster-operator/pkg/authorization"
+	hcoutil "github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
+)
+
+const (
+	secretName = "hco-bearer-auth"
+)
+
+type secretReconciler struct {
+	theSecret *corev1.Secret
+}
+
+func newSecretReconciler(namespace string, owner metav1.OwnerReference) *secretReconciler {
+	return &secretReconciler{
+		theSecret: NewSecret(namespace, owner),
+	}
+}
+
+func (r *secretReconciler) Kind() string {
+	return "Secret"
+}
+
+func (r *secretReconciler) ResourceName() string {
+	return secretName
+}
+
+func (r *secretReconciler) GetFullResource() client.Object {
+	return r.theSecret.DeepCopy()
+}
+
+func (r *secretReconciler) EmptyObject() client.Object {
+	return &corev1.Secret{}
+}
+
+func (r *secretReconciler) UpdateExistingResource(ctx context.Context, cl client.Client, resource client.Object, logger logr.Logger) (client.Object, bool, error) {
+	found := resource.(*corev1.Secret)
+	modified := false
+
+	token, err := authorization.CreateToken()
+	if err != nil {
+		logger.Info("failed to create bearer token", "message", err)
+		token = ""
+	}
+
+	if found.Data["token"] == nil || string(found.Data["token"]) != token {
+		found.StringData = map[string]string{
+			"token": token,
+		}
+		modified = true
+	}
+
+	modified = updateCommonDetails(&r.theSecret.ObjectMeta, &found.ObjectMeta) || modified
+
+	if modified {
+		err := cl.Update(ctx, found)
+		if err != nil {
+			logger.Error(err, "failed to update the Secret")
+			return nil, false, err
+		}
+	}
+
+	return found, modified, nil
+}
+
+func NewSecret(namespace string, owner metav1.OwnerReference) *corev1.Secret {
+	token, err := authorization.CreateToken()
+	if err != nil {
+		logger.Info("failed to create bearer token", "message", err)
+		token = ""
```

Comment:
Should we create the secret w/o the token?

Reply:
I didn't want to throw an error and crash the whole pod
nor not create the secret at all, and disallow the access to the endpoint

I thought about generating a secret and keeping it in memory, maybe that's better, wdyt?

@nunnatsa

Reply:
I don't think we need to kill the pod. Just try again in the next reconcile loop. Will ot work?

Reply:
I wonder if there is just some configuration where the `serviceaccount/token` is not populated in the pod, it will not exist in any further reconcilations